drivers/can/can_handlers.c

```diff
@@ -35,29 +37,13 @@ Z_SYSCALL_HANDLER(can_attach_msgq, dev, msgq, filter) {
 
 	Z_OOPS(Z_SYSCALL_MEMORY_READ((struct zcan_filter *)filter,
 				     sizeof(struct zcan_filter)));
-	Z_OOPS(Z_SYSCALL_MEMORY_WRITE((struct k_msgq *)msgq,
-				      sizeof(struct k_msgq)));
-	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(((struct k_msgq *)msgq)->buffer_start,
```

Comment:
Why is this check not needed?

Reply:
because user mode can *never* write directly to a k_msgq. They are full fledged kernel objects. The call I replaced these with checks that this is a valid k_msgq, properly initialized, and that the caller has permissions on it.